taskt/Core/Script.cs

```diff
@@ -234,13 +233,16 @@ public string GetDisplayValue(string requiredProperty = "")
                     default:
                         return (string)VariableValue;
                 }
-              
             }
             else if(VariableValue is DataTable)
             {
-                DataTable dataTable = (DataTable)VariableValue;              
+                DataTable dataTable = (DataTable)VariableValue;
                 var dataRow = dataTable.Rows[CurrentPosition];
-                return Newtonsoft.Json.JsonConvert.SerializeObject(dataRow.ItemArray);            
+                return Newtonsoft.Json.JsonConvert.SerializeObject(dataRow.ItemArray);
+            }
+            else if(VariableValue.GetType() != typeof(string))
+            {
+                return VariableValue.ToString();

```

Comment:
Won't this produce an error if you're returning a data type that isn't one of the ones above but also can't be converted into a string?

Reply:
`ToString` is defined in Object so if a specific object doesn't override it then it'll execute `Object.ToString` which by default returns the qualified class name of the object (i.e if you do it on a data row it'll return `System.Data.DataRow`).

While it isn't the most complete solution this change prevents exceptions from happening in ConvertToUserVariable such as in this line: https://github.com/CameronHerwig/taskt/blob/213f9fb7c0c2659f4c643d3d4344245421c17aca/taskt/Core/ExtensionMethods.cs#L201

and gives us a way to return string representations for values like Bools and Ints without causing an exception.

Reply:
Okay then. Sounds good, thanks